app/assets/javascripts/scroll-into-view.js

```diff
@@ -0,0 +1,19 @@
+(function () {
+  document.addEventListener("DOMContentLoaded", function () {
+    var input = document.querySelector(".govspeak-preview-input form"),
+      output = document.querySelector(".govspeak-preview-output");
+
+    if (!input || !output) return;
+    if (output.dataset.ouptputIsNil === "true") return;
+
+    input.scrollIntoView({
+      behavior: setTransitionAnimation(),
+    });
+  });
+
+  function setTransitionAnimation() {
+    return window.matchMedia("(prefers-reduced-motion: reduce)").matches
```

Comment:
💚

Reply:
I would be tempted to add a class with the JS and move this to the [CSS](https://codepen.io/ericwbailey/pen/PWJPrW). 

(Side thought: I feel like this would be nice to have something generic in either the Gem or the DS. So we can apply or use this more liberally using something like `govuk-animation`)

Reply:
Thanks Chris, I see what you mean. I think I could add a class for something like `scroll-behavior: smooth` https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-behavior#browser_compatibility and add the `prefers-reduced-motion` media feature query https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion. However, I think browser support could be an issue for IE11 so would need to investigate some more.

For now, i'll keep with JavaScript (also because i'm using JS to handle the scroll behaviour seems nice to keep all this together) but will definitely bear this in mind as something we may want to do in future (also, yes, could be useful to scope out use cases in the Gem for something like this).